### PR TITLE
Add Output.concat to Python

### LINKED ISSUE #2366: Add 'concat' and 'interpolate' equivalencies for Python

### PR DESCRIPTION
**Pull request by lukehoban**
Fixes #2366

### CODE REVIEW

**Comment by lukehoban** [comment]
Going to go ahead and merge as-is - but @CyrusNajmabadi let me know if you have any post-hoc review feedback.

File: sdk/python/lib/pulumi/output.py [hunk 1]
```diff
@@ -317,3 +317,21 @@ def all(*args: List[Input[T]]) -> 'Output[List[T]]':
...
+
+            url = Output.concat("http://", server.hostname, ":", loadBalancer.port)
+
+        :param List[Input[str]] args: A list of string Inputs to concatenate.
+        :return: A concatenated output string.
+        :rtype: Output[str]
+        """
+
+        transformed_items = [Output.from_input(v) for v in args]
+        return Output.all(*transformed_items).apply("".join)
```
Comment (CyrusNajmabadi) [inline sdk/python/lib/pulumi/output.py]: i take it you never figured out hte issue from yesteday?

File: sdk/python/lib/test/langhost/output_all/__main__.py [hunk 2]
```diff
@@ -43,3 +43,7 @@ def assert_eq(lhs, rhs):
 # Output.all combines a list of outputs into an output of a list.
 resSum = Output.all(res1.number, res2.number).apply(lambda l: l[0] + l[1])
 FinalResource("testResource3", resSum)
+
+# Test additional Output helpers
+hello_world = Output.concat("Hello ", Output.from_input("world!")).apply(lambda s: assert_eq(s, "Hello world!"))
```
Comment (CyrusNajmabadi) [inline sdk/python/lib/test/langhost/output_all/__main__.py]: cute!

File: sdk/python/lib/pulumi/output.py (hunk 1 above)
Comment (lukehoban) [reply sdk/python/lib/pulumi/output.py]: Unfortunately no - and time-boxed investigating it.  I'm assuming at some point we'll see it reported via some more user-facing repro, and will need to investigate more deeply.

**Pull request merged**